Cython/Utility/CythonFunction.c

```diff
@@ -362,6 +364,18 @@ __Pyx_CyFunction_get_annotations(__pyx_CyFunctionObject *op, CYTHON_UNUSED void
     return result;
 }
 
+static PyObject *
+__Pyx_CyFunction_get_is_coroutine(__pyx_CyFunctionObject *op, CYTHON_UNUSED void *context) {
+    PyObject* result = Py_None;
+    if (op->flags & __Pyx_CYFUNCTION_COROUTINE) {
+        PyObject *module = PyImport_ImportModule("asyncio.coroutines");
+        result = PyObject_GetAttrString(module, "_is_coroutine");
```

Comment:
This case leaks a reference with the `Py_INCREF()` below, because it already is an owned reference. (Also, if the lookup fails, then the result is `NULL` and you get a crash on the incref.)

Also, it's more efficient to use Python strings directly instead of C strings for the lookup:
```suggestion
        result = __Pyx_PyObject_GetAttrStr(module, PYIDENT("_is_coroutine"));
```
This requires an additional declaration at the beginning of this utility code section (~line 95) to also include this specific helper function:
```
//@requires: PyObjectGetAttrStr
```

You can also use `PYIDENT("asyncio.coroutines")` and then call `PyImport_ImportModuleLevelObject()`, at least in Py3. Py2 doesn't have that module anyway. And older Py3s don't have `._is_coroutine`. Check `PY_VERSION_HEX` to make it a Py3.5(?)-only feature.

Reply:
Oh! Thanks for letting me know. :) Will amend, and add the PY_VERSION_HEX checks.

Reply:
FYI... Double checked and on v3.4.2rc1 `_is_coroutine` check was introduced, but it was just a True/False value. On v3.5.3rc1 the marker object was introduced. I'll add the macros to run for both versions.